outputs.tf

```diff
@@ -42,67 +42,67 @@ output "launch_template_latest_version" {
 
 output "autoscaling_group_id" {
   description = "The autoscaling group id"
-  value       = element(concat(aws_autoscaling_group.this.*.id, [""]), 0)
+  value       = try(aws_autoscaling_group.this[0].id, "")
 }
 
 output "autoscaling_group_name" {
   description = "The autoscaling group name"
-  value       = element(concat(aws_autoscaling_group.this.*.name, [""]), 0)
+  value       = try(aws_autoscaling_group.this[0].name, "")
 }
 
 output "autoscaling_group_arn" {
   description = "The ARN for this AutoScaling Group"
-  value       = element(concat(aws_autoscaling_group.this.*.arn, [""]), 0)
+  value       = try(aws_autoscaling_group.this[0].arn, "")
 }
 
 output "autoscaling_group_min_size" {
   description = "The minimum size of the autoscale group"
-  value       = element(concat(aws_autoscaling_group.this.*.min_size, [""]), 0)
+  value       = try(aws_autoscaling_group.this[0].min_size, "")
 }
 
 output "autoscaling_group_max_size" {
   description = "The maximum size of the autoscale group"
-  value       = element(concat(aws_autoscaling_group.this.*.max_size, [""]), 0)
+  value       = try(aws_autoscaling_group.this[0].max_size, "")
 }
 
 output "autoscaling_group_desired_capacity" {
   description = "The number of Amazon EC2 instances that should be running in the group"
-  value       = element(concat(aws_autoscaling_group.this.*.desired_capacity, [""]), 0)
+  value       = try(aws_autoscaling_group.this[0].desired_capacity, "")
 }
 
 output "autoscaling_group_default_cooldown" {
   description = "Time between a scaling activity and the succeeding scaling activity"
-  value       = element(concat(aws_autoscaling_group.this.*.default_cooldown, [""]), 0)
+  value       = try(aws_autoscaling_group.this[0].default_cooldown, "")
 }
 
 output "autoscaling_group_health_check_grace_period" {
   description = "Time after instance comes into service before checking health"
-  value       = element(concat(aws_autoscaling_group.this.*.health_check_grace_period, [""]), 0)
+  value       = try(aws_autoscaling_group.this[0].health_check_grace_period, "")
 }
 
 output "autoscaling_group_health_check_type" {
   description = "EC2 or ELB. Controls how health checking is done"
-  value       = element(concat(aws_autoscaling_group.this.*.health_check_type, [""]), 0)
+  value       = try(aws_autoscaling_group.this[0].health_check_type, "")
 }
 
 output "autoscaling_group_availability_zones" {
   description = "The availability zones of the autoscale group"
-  value       = element(concat(aws_autoscaling_group.this.*.availability_zones, [""]), 0)
+  value       = try(aws_autoscaling_group.this[0].availability_zones, [])
 }
 
 output "autoscaling_group_vpc_zone_identifier" {
   description = "The VPC zone identifier"
-  value       = element(concat(aws_autoscaling_group.this.*.vpc_zone_identifier, [""]), 0)
+  value       = try(aws_autoscaling_group.this[0].vpc_zone_identifier, "")
```

Comment:
`vpc_zone_identifier` is a list. Don't ask me how I know :))

Reply:
fixed in https://github.com/terraform-aws-modules/terraform-aws-autoscaling/pull/169/commits/6123a8730f93da34e121bc7d2d9063c27a43287c